### ISSUE TITLE
SmarterCSV doesn't parse the second line of StringIO

### ISSUE THREAD

**Issue by iSimar** [tilo/smarter_csv#200]
## Define a sample CSV based StringIO
```
csv = StringIO.new(
  <<~CSV,
        First Name,Last Name
        A,B
        C,D
  CSV
)
```

## Invoke SmarterCSV.process
```
SmarterCSV.process(csv)  # output {:first_name=>"A", :last_name=>"B"}
```

## Expected output
```
[ {:first_name=>"A", :last_name=>"B"}, {:first_name=>"C", :last_name=>"D"} ]
```

**Comment by hut8** [comment]
Probably the same as #197

**Comment by tilo** [comment]
@iSimar what operating system are you using? Windows? Mac? Linux?

**Comment by iSimar** [comment]
@hut8 Yes it worked with 1.6.1
@tilo Mac

**Comment by tilo** [comment]
@iSimar something is very odd 🤔 

When I try this on my M1 MacBook, it works:

```
3.0.0 :008 > require 'stringio'
 => true
3.0.0 :009 > csv = StringIO.new(
3.0.0 :010">   <<~CSV,
3.0.0 :011">         First Name,Last Name
3.0.0 :012">         A,B
3.0.0 :013">         C,D
3.0.0 :014 >   CSV
3.0.0 :015 > )
 => #<StringIO:0x00007f7a6780e2b8>
3.0.0 :016 > require 'smarter_csv'
 => true
3.0.0 :017 > SmarterCSV.process(csv)
 => [{:first_name=>"A", :last_name=>"B"}, {:first_name=>"C", :last_name=>"D"}]
 ```
 
I'll try to find an x86 macbook to reproduce

**Comment by tilo** [comment]
OK, so on an x86 MacBook with version 1.7.0 it also reads the CSV correctly for me 🤔 

```
>  SmarterCSV.process(csv)
 => [{:first_name=>"A", :last_name=>"B"}, {:first_name=>"C", :last_name=>"D"}]

> `uname -a`
 => "Darwin OldLaptop 20.6.0 Darwin Kernel Version 20.6.0: Tue Apr 19 21:04:45 PDT 2022; root:xnu-7195.141.29~1/RELEASE_X86_64 x86_64\n"
```

**Comment by tilo** [comment]
Bugfix release 1.7.1 was just published

Please re-evaluate and update this issue if this fixes the problem

**Comment by tilo** [comment]
TL;DR: the issue only showed up when `smarter_csv` was used in a Rails project.

the issue is fixed in 1.7.1

**Comment by hut8** [comment]
@tilo thanks so much with your hard work debugging. It is much appreciated and does not go unnoticed 😄

**Comment by tilo** [comment]
> @tilo thanks so much with your hard work debugging. It is much appreciated and does not go unnoticed 😄

🙇 🙇 🙇
